frontend/hooks/useAuth.js

```diff
@@ -1,12 +1,24 @@
-import React, { createContext, useContext, useState } from "react";
+import React, { createContext, useContext, useState, useEffect } from "react";
 import PropTypes from "prop-types";
 
-import { login, logout } from "root/api/auth";
+import { login, logout, getSession } from "root/api/auth";
 
 export const AuthContext = createContext();
 
-export const AuthProvider = ({ initialUser, children }) => {
-  const [user, setUser] = useState(initialUser);
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [fetchingInitialUser, setFetchingInitialUser] = useState(true);
+
+  useEffect(() => {
+    fetchInitialUser();
+  }, []);
+
+  async function fetchInitialUser() {
+    const response = await getSession();
+
+    setUser(response.data.user);
```

Comment:
If the user is not logged, what will be the response here?

Reply:
I'll add more clear code to this.